core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+            android_sdk_version=android_sdk_version,
+            android_device_model=android_device_model, entry_point=entry_point,
+            entry_point_topic_id=entry_point_topic_id,
+            entry_point_exploration_id=entry_point_exploration_id,
+            entry_point_story_id=entry_point_story_id,
+            entry_point_subtopic_id=entry_point_subtopic_id,
+            text_language_code=text_language_code,
+            audio_language_code=audio_language_code,
+            android_report_info=android_report_info,
+            android_report_info_schema_version=android_schema_version,
+            web_report_info=web_report_info,
+            web_report_info_schema_version=web_schema_version)
+        report_entity.update_timestamps()
+        report_entity.put()
+        return entity_id
+
+    @classmethod
+    def _generate_id(cls, platform, submitted_on_sec):
+        """Generates key for the instance of AppFeedbackReportModel class in the
+        required format with the arguments provided.
+
+        Args:
+            platform: str. The platform the user is the report from.
+            submitted_on_sec: float. The timestamp that the report was submitted
+                on, in seconds since epoch (in UTC).
+
+        Returns:
+            str. The generated ID for this entity using platform,
+            submitted_on_sec, and a random string, of the form
+            '[platform].[submitted_on_sec].[random hash]'.
+        """
+        for _ in python_utils.RANGE(base_models.MAX_RETRIES):
+            random_hash = utils.convert_to_hash(
+                python_utils.UNICODE(
+                    utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH)
+            new_id = '%s.%s.%s' % (platform, submitted_on_sec, random_hash)
+            if not cls.get_by_id(new_id):
+                return new_id
+        raise Exception(
+            'The id generator for AppFeedbackReportModel is producing too '
+            'many collisions.')
+
+    @classmethod
+    def scrub_report(cls, report_id, scrubbed_by):
+        """Scrubs the instance of AppFeedbackReportModel with given ID, removing
+        any user-entered input in the entity.
+
+        Args:
+            report_id: str. The id of the model entity to scrub.
+            scrubbed_by: str. The id of the user or cron job that is intiating
+                scrubbing this report.
+        """
+        cls._scrub_report_in_transaction(report_id, scrubbed_by)
+
+    @classmethod
+    @transaction_services.run_in_transaction_wrapper
+    def _scrub_report_in_transaction(cls, report_id, scrubbed_by):
+        """See scrub_report for general documentaion of what this method does.
+        It's only safe to call this method from within a transaction.
+
+        Args:
+            report_id: str. The id of the model entity to scrub.
+            scrubbed_by: str. The id of the user or cron job that is intiating
```

Comment:
Cron jobs do not typically have an ID. Maybe give more detail about what the value would be in that case.

Reply:
Done